add docs for host field in .status (#554)
argoproj-labs/argocd-operator: docs/usage/ingress.md

[install_olm]:../install/olm.md
[docs_argo]:https://argoproj.github.io/argo-cd/getting_started/#creating-apps-via-cli

> ### Host URL for Ingress in Argo CD Status

Reviewer: ```suggestion
### Host for Ingress in Argo CD Status
```

> When setting up access to Argo CD via a Route or an Ingress, one can easily retrieve the URL used for accessing the Argo CD installation through the ArgoCD Operand's `.status` field. To expose the `.host` field, run `kubectl edit argocd argocd` and then edit the Argo CD instance server to have ingress enabled as `true`, like so: 

Reviewer: ```suggestion
When setting up access to Argo CD via an Ingress, one can easily retrieve hostnames used for accessing the Argo CD installation through the ArgoCD Operand's `status` field. To expose the `host` field, run `kubectl edit argocd argocd` and then edit the Argo CD instance server to have ingress enabled as `true`, like so: 
```

```yaml
server:
    autoscale:
      enabled: false
    grpc:
      ingress:
        enabled: false

>     ingress:

Reviewer: Setting server.grpc.ingress.enabled to`false` is not wrong. The default seems to be `false`. If it is set to `true`, the operator will create an Ingress for grpc protocol to reach the ArgoCD server which is used by CLI. The hostname of the Ingress will be shown in the `status`. The question is should the example illustrate the rpc ingress with other ingresses.

>         enabled: false

Reviewer: Should this be `true` or `false`?

Author: that's unrelated I'm pretty sure, it's usually kept as false.

      enabled: true
    route:
      enabled: false
    service:
      type: ""
  tls:
    ca: {}
```

> If an ingress is found, you host URL can now be accessed by inspecting your Argo CD instance. This data could be the hostname and/or the IP address, depending on what data is available. It will look like the following: 

Reviewer: ```suggestion
If an ingress is found, hostname(s) of the ingress can now be accessed by inspecting your Argo CD instance. This data could be hostname(s) and/or the IP address(es), depending on what data is available. It will look like the following: 
```

```yaml
status:
  applicationController: Running
  dex: Running
  host: 172.24.0.7
  phase: Available
  redis: Running
  repo: Running
  server: Running
  ssoConfig: Unknown
```

> If both Routes and Ingress are enabled in the Argo CD spec and a route is available, the data for the Route will be prioritized over the data for Ingress. 

Reviewer: ```suggestion
If both Route and Ingress are enabled in the Argo CD spec and a route is available, the status for the Route will be prioritized over the Ingress's. In the case, the `host` for the Ingress is not shown in the `status`.
```

> If the status of `.host` is pending, unlike with Routes, this will not affect the overall status of the Operand.

Reviewer: ```suggestion
Unlike with Routes, an Ingress does not go to pending status.  Hence, this will not affect the overall status of the Operand.
```
